npm/src/debug-service/debug-service.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { DebugService } from "../../lib/node/qsc_wasm.cjs";
+import { eventStringToMsg } from "../compiler/common.js";
+import { IQscEventTarget, QscEvents, makeEvent } from "../compiler/events.js";
+import { log } from "../log.js";
+import { IServiceProxy } from "../worker-proxy.js";
+import { BreakpointSpan, StackFrame } from "./types.js";
+type QscWasm = typeof import("../../lib/node/qsc_wasm.cjs");
+
+// These need to be async/promise results for when communicating across a WebWorker, however
+// for running the compiler in the same thread the result will be synchronous (a resolved promise).
+export interface IDebugService {
+  loadSource(path: string, source: string): Promise<boolean>;
+  getBreakpoints(path: string): Promise<BreakpointSpan[]>;
+  getStackFrames(): Promise<StackFrame[]>;
+  evalContinue(
+    bps: Uint32Array,
```

Comment:
Interesting. I can see why `wasm_bindgen` generated `Uint32Array` as a type, but I generally don't expect to see it on a JS method - `number[]` is more common. To make this a more cohesive interface I suggest having this take a `number[]` and doing the conversion in the implementation in this same file. (You're already doing the conversion today, just up in the VS Code layer in ` get_breakpoint_ids`).